### PR TITLE
Updating documentation

### PR DESCRIPTION
**Pull request by manishgarg1**
#### Description of changes

Some document restructuring to set it up for future improvements.

### CODE REVIEW

File: FAQ.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+http://dev.office.com/fabric
+
+## *Q. Can I contribute to Office UI Fabric React?*
+
+Office UI Fabric React is an open-source distributed project. Lots of developers from inside and outside Microsoft contribute to this project. We highly encourage contributions. There is a core team that works hard to stay on top of the issues and pull requests.
+
+## *Q. Why should I contribute?*
+
+Office UI Fabric React is an open-source distributed project. If you are building a new app inside O365 or Microsoft, this project will potentially save you lots of time, resources and headaches. We encourage all developers who use Fabric React to return a small amount to the community. If you notice bugs, we encourage you to not only open an issue on it but also help fix it. If you have a component that you believe others should use, feel motivated and encouraged to contribute that component back to the community. Sharing is caring. Many developers contribute outside their primary job responsibilities. Additionally, there is plenty to learn. We use a lot of best practices used in the industry. Learning those can  help you in your career.
```
[the remark this answers is not in the document]
Comment (manishgarg1) [reply FAQ.md]: Big thanx for reviewing @Jahnp.

File: ghdocs/BestPractices/Testing.md [hunk 2]
```diff
@@ -4,7 +4,7 @@
 
 Our tests are built using [Jest](https://facebook.github.io/jest/). This allows us to run tests in a node environment, and simulates the browser using jsdom.
 
-For snapshot testing, we use `react-test-renderer` and Jest apis.
+For snapshot testing, we use `react-test-renderer` and `Jest apis`.
```
[the remark this answers is not in the document]
Comment (manishgarg1) [reply ghdocs/BestPractices/Testing.md]: @jahnp, for some of these docs, I am just moving them at this point. Will then work on them individually once this PR is merged.

---
In reply to: [180287991](https://github.com/OfficeDev/office-ui-fabric-react/pull/4490#discussion_r180287991) [](ancestors = 180287991)

File: ghdocs/Contributing/NewComponent.md [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+2. Run `npm run create-component -- --name ExcitingNewComponent`.
+
+#### Creating a component manually
+1. From the root of the project, navigate to `packages/experiments/src/components/`
+2. Create a new folder here with your component name
+  * Our components use the Pascal Case naming convention.
+  * In this example the component will be called `ExcitingNewComponent`.
+3. Create a `ExcitingNewComponent.types.ts` file that will contain an interface for your component props
+
+  * @todo: review
```
Comment (manishgarg1) [inline ghdocs/Contributing/NewComponent.md]: >todo [](start = 5, length = 4)

Resolve before merging

File: ghdocs/Contributing/NewComponent.md [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+  * Import React
+    * `import * as React from 'react'`;
+  * Import ExcitingNewComponent
+    * `import { ExcitingNewComponent } from './ExcitingNewComponent'`;
+    * Note: This class and file don't exist yet but they will during step 4 of this small tutorial.
+  * Create an empty interface `IExcitingNewComponent`
```
Comment (dzearing) [inline ghdocs/Contributing/NewComponent.md]: Create an empty interface (which may contain the public imperative API surface, such as `focus()`.

File: ghdocs/Contributing/NewComponent.md [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+
+```ts
+import * as React from 'react';
+import { ExcitingNewComponent } from './ExcitingNewComponent';
+import { IStyle } from '../../Styling';
+
+export interface IExcitingNewComponent {
+}
+
+export interface IExcitingNewComponentProps extends React.Props<ExcitingNewComponent> {
```
Comment (dzearing) [inline ghdocs/Contributing/NewComponent.md]: the ts file should import the types from the file.

File: ghdocs/Contributing/NewComponent.md [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+  /**
+  * All props for your component are to be defined here.
+  */
+  componentRef?: (component: IExcitingNewComponent) => void;
```
Comment (dzearing) [inline ghdocs/Contributing/NewComponent.md]: this should be moved up into the .types description, explaining this is how customers should access the imperative API surface (through `componentRef`)

File: ghdocs/BestPractices/Testing.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (manishgarg1) [reply ghdocs/BestPractices/Testing.md]: Very little here and there where the meaning of the content won't change.

---
In reply to: [180502872](https://github.com/OfficeDev/office-ui-fabric-react/pull/4490#discussion_r180502872) [](ancestors = 180502872)

**Pull request merged**